src/components/Checkbox/Checkbox.st.css

```diff
@@ -1,47 +1,102 @@
 :import {
-  -st-from: "wix-ui-core/dist/src/components/Checkbox/CheckboxStyle.st.css";
+  -st-from: "wix-ui-core/dist/src/components/Checkbox/Checkbox.st.css";
   -st-default: Checkbox;
 }
 
 :import {
-  -st-from: "wix-ui-core/dist/src/components/Checkbox/Checkbox.st.css";
-  -st-default: BaseCheckbox;
+  -st-from: "../../palette.st.css";
+  -st-named: backgroundPrimary, backgroundSecondary, mainMutedBackground, mainMuted, main, mainHover, dangerMutedHover, danger, disabledFields, dividerColor;
 }
 
-:import {
-  -st-from: "../../palette.st.css";
-  -st-named: mainMuted, main, mainHover, disabled, successMutedHover, dangerMutedHover, success, danger, successHover, dangerHover, dividerColor, disabled;
+:vars {
+  large: 20px;
+  medium: 14px;
+  iconTick: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxNiAxNiI+PHBhdGggZmlsbD0iI2ZmZiIgZD0iTTExLjggNEw2LjUgOS4yIDQuMiA3IDMgOC4ybDMuNSAzLjZMMTMgNS4yIi8+PC9zdmc+);
+  iconTickDisabled: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxNiAxNiI+PHBhdGggZmlsbD0iI2M4YzhjOCIgZD0iTTExLjggNEw2LjUgOS4yIDQuMiA3IDMgOC4ybDMuNSAzLjZMMTMgNS4yIi8+PC9zdmc+);
+  iconIntermediate: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxNiAxNiI+PHBhdGggZmlsbD0iI2ZmZmZmZiIgZD0iTTQgN2g4djJINHoiLz48L3N2Zz4=);
+  iconIntermediateDisabled: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxNiAxNiI+PHBhdGggZmlsbD0iI2M4YzhjOCIgZD0iTTQgN2g4djJINHoiLz48L3N2Zz4=);
 }
 
 .root {
-  -st-extends: BaseCheckbox;
-  -st-states: error, success, x-small, small;
-  -st-mixin: Checkbox(
-    rootWidth initial,
-    rootHeight initial,
-
-    outerLabelWidth 45px,
-    outerLabelHeight 24px,
-
-    innerLabelWidth 21px,
-    innerLabelHeight 22px,
-    innerLabelBackgroundColor white,
-
-    backgroundColor value(mainMuted),
-    backgroundColorChecked value(main),
-    backgroundColorDisabled value(disabled),
-    backgroundColorHover value(mainHover),
-    backgroundColorFocus value(mainHover),
-
-    color value(mainMuted),
-    colorChecked value(main),
-    colorDisabled value(dividerColor),
-    colorCheckedDisabled value(disabled),
-    colorHover value(mainHover),
-    colorFocus value(mainHover),
-
-    toggleIconWidth 8px,
-    toggleIconHeight 6px,
-    toggleIconDisplay block
-  );
+  -st-extends: Checkbox;
+  -st-states: large, medium;
```

Comment:
I think we can drop the large design. It is not defined in the design system AFAIK.